### DISCUSSION TITLE
Good way to link two stores from contexts

### DISCUSSION THREAD

**Discussion by hellozyemlya** [pmndrs/zustand#2273] [Q&A] [answered]
I have two stores. 

One store provides one portion of reusable state, created via factory method using createStore and pushed to context(Store1Ctx, and Store1CtxProvider).

Another store exposed in same way using createStore and separate context(Store2Ctx, Store2CtxProvider), but it is optional and requires some constant updates from first store.

Is there a good way to link that stores? For now, I just get first store in Store2CtxProvider and subscribe for changes like useContext(Store1Ctx).subscribe((new, old) => ...) in effect. Is that fine, or maybe there is another way to combine both stores dynamically?

**Comment by dai-shi** [comment, marked as answer]
In general, two store should be isolated, and if you need to combine them, having one store would be a better solution.
That said, if having two stores and you want keep in sync, `useEffect` or `.subscribe()` might work. (It's not efficient, but can't be helped. Be careful for infinite loop.)
